src/main/src/test/java/org/geoserver/test/GeoServerSystemTestSupport.java

```diff
@@ -164,9 +165,33 @@ protected SystemTestData createTestData() throws Exception {
     /**
      * credentials for mock requests
      */
-    protected String username, password; 
+    protected String username, password;
+
+    /**
+     * Cached dispatcher, it has its own app context, so it's expensive to build
+     */
+    protected static DispatcherServlet dispatcher;
+
+    protected boolean isFullInitRequired() {
+        return false;
+    }
 
     protected final void setUp(SystemTestData testData) throws Exception {
+        // speed up xpath evaluations
+        try {
+            // see
+            // http://stackoverflow.com/questions/6340802/java-xpath-apache-jaxp-implementation-performance
+            Class.forName("com.sun.org.apache.xml.internal.dtm.ref.DTMManagerDefault");
+            System.setProperty("com.sun.org.apache.xml.internal.dtm.DTMManager",
+                    "com.sun.org.apache.xml.internal.dtm.ref.DTMManagerDefault");
+        } catch (Exception e) {
+            // ignore on VM where this optimization does not apply
+        }
+
+        // disable security manager to speed up tests, we are spending a lot of time in privileged
+        // blocks
+        System.setSecurityManager(null);
```

Comment:
Worth noticing that Oracle is aware of the issue and there is a JEP for JDK 9 to speed up security checks in doPrivileged.